Integration Docs (#73)
dappforce/grillchat: INTEGRATION.md

# How to Integrate Grill.chat to your app 🤝

This guide will help you to integrate Grill.chat to your app, with all the options available.

## Iframe Integration

The easiest way to integrate Grill.chat to your app is to use the iframe integration. It's a simple HTML tag that you can add to your app, and it will render the chat in an iframe.

### 1. Add Iframe tag to your app

> Add the following HTML tag to your app, where you want to render the chat and style it however you want.

Reviewer: ```suggestion
Add the following HTML tag to your app in the place that you want to render the chat, and style it however you want.
```

```html
<iframe allow="clipboard-write" src="https://x.grill.chat"></iframe>
```

### 2. Customize the `src` link

> The `src` link can be customized to your needs. Below includes the list of options that you can use to customize the chat.

Reviewer: ```suggestion
The `src` link can be customized to your needs. Below is the list of options that you can use to customize the chat.
```

#### 2.1. Space Options

> Grill.chat home page contains list of topics for user to choose from. This topics are listed from all the posts in a subsocial space.

Reviewer: ```suggestion
The Grill.chat home page contains a list of topics for users to choose from. On the back-end, these topics are the posts in a particular space on Subsocial.
```

> For example, if you create a space with ID `1002`, then you can set the iframe `src` to `https://grill.chat/1002` to only show topics from that space.

Reviewer: ```suggestion
For example, if you create a space with the ID `1002`, you can set the iframe `src` to `https://grill.chat/1002` and it will only show the topics from that space.
```

#### 2.2. Page Options

> Grill.chat has 2 pages that can be the start point.

Reviewer: ```suggestion
Grill.chat has 2 pages that can be the starting screen.
```

1. Home page

>    It contains all of the topics in a subsocial space that you set in the `src` link. You can read more in [Space Options](#21-space-options).

Reviewer: ```suggestion
   It contains all of the topics in the Subsocial space that you set in the `src` link. You can read more in [Space Options](#21-space-options).
```

   ```
   https://grill.chat/[spaceId]
   ```

2. Chat Page

>    You can also choose to have your user automatically opens a chat room. This is useful if you want to have a specific topic for your user to discuss.

Reviewer: ```suggestion
   You can also choose to have your users automatically enter a chat room. This is useful if you want to have a specific topic for your user to discuss.
```

   ```
   https://grill.chat/[spaceId]/c/[topicId]
   ```

#### 2.3. Other options

> You can also customize the `src` link with query parameters. Below includes the list of query parameters that you can use to customize the chat.

Reviewer: ```suggestion
You can also customize the `src` link with query parameters. Below is the list of query parameters that you can use to customize the chat.
```

generate table

| Name  | Description                                                                                                                                                         |
| ----- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | theme | Theme of the chat. Available options: `light` or `dark`. If not provided, it will use user's system preferences or his last selected theme when accessed Grill.chat |

Reviewer: ```suggestion
| theme | Theme of the chat. The available options are `light` or `dark`. If a theme is not provided, it will use the user's system preferences, or the theme that was selected when the user last accessed Grill.chat. |
```

> | order | Specifies the order of the chat in home page. Pass multiple topic ids in `order` separated by comma. The other chats will be ordered based on last message          |

Reviewer: ```suggestion
| order | Specifies the order of the various topics/chats on the home page. Pass multiple topic IDs in the `order` that you want, separated by a comma. Any other chats will be ordered based on the last message sent.         |
```
